crates/astria-core/src/protocol/transaction/v1alpha1/mod.rs

```diff
@@ -10,6 +10,9 @@ use super::raw;
 pub mod action;
 pub use action::Action;
 
+pub const UNSIGNED_TRANSACTION_TYPE_URL: &str =
+    "/astria.protocol.transaction.v1alpha1.UnsignedTransaction";
```

Comment:
I am not generally opposed to doing this (to save some allocations), but we must ensure that const matches the generated type URL of the `prost::Name` trait. So either use `UnsignedTransaction::type_url()` everywhere, or alternatively use the hardcoded value here, but add a unit test like below.

```rust
#[test]
fn hardcoded_type_url_matches_generated_trait() {
    assert_eq!(UNSIGNED_TRANSACTION_TYPE_URL, &UnsignedTransaction::type_url(),);
}
```

Read my other comments re the use of `Name::type_url()` with this in mind - either use the trait method everywhere, or ensure that the const always stays valid.

Reply:
updated to use `type_url()`